rs/nervous_system/integration_tests/tests/sns_upgrade_test_utils.rs

```diff
@@ -109,19 +62,117 @@ pub async fn test_sns_upgrade(sns_canisters_to_upgrade: Vec<SnsCanisterType>) {
     )
     .await;
 
-    // Every canister we are testing has two upgrades.  We are just making sure the counts match
-    for canister_type in &sns_canisters_to_upgrade {
-        eprintln!(
-            "1st upgrade_sns_to_next_version_and_assert_change {:?} ...",
-            canister_type
-        );
-        sns::upgrade_sns_to_next_version_and_assert_change(&pocket_ic, &sns, *canister_type).await;
+    eprintln!("Adding all WASMs ...");
+    let latest_sns_version = {
+        let mut latest_version = initial_sns_version;
+
+        for canister_type in &sns_canisters_to_upgrade {
+            eprintln!("modify_and_add_wasm for {:?} ...", canister_type);
+            latest_version =
+                nns::sns_wasm::modify_and_add_wasm(&pocket_ic, latest_version, *canister_type, 1)
+                    .await;
+        }
+
+        latest_version
+    };
+
+    {
+        eprintln!("wait for the upgrade steps to be refreshed ...");
+        let latest_sns_version_hash = hash_sns_wasms(&latest_sns_version);
+        await_with_timeout(
+            &pocket_ic,
+            UPGRADE_STEPS_INTERVAL_REFRESH_BACKOFF_SECONDS,
+            |pocket_ic| async {
+                sns::governance::try_get_upgrade_journal(pocket_ic, sns.governance.canister_id)
+                    .await
+                    .ok()
+                    .and_then(|journal| journal.upgrade_steps)
+                    .and_then(|upgrade_steps| upgrade_steps.versions.last().cloned())
+            },
+            &Some(latest_sns_version_hash.clone()),
+        )
+        .await
+        .unwrap();
+
+        eprintln!("advance the target version to the latest version. ...");
```

Comment:
Let's either turn this into comments or evaluate the test runtime with and without these prints.

Reply:
Without printing: 149.0s
With printing: 143.6s

So with printing is somehow faster in my test. I assume this is noise. Regardless, I see no evidence that the printing is adding any substantial slowdown